thirdparty-scms/ado/functions.sh

```diff
@@ -0,0 +1,169 @@
+    --branch master --description "Pipeline SDLF for ${PREFIX}-${REPOSITORY}" \
+    --folder-path "sdlf" --yml-path azure-pipelines.yml \
+    --repository "${PREFIX}-${REPOSITORY}" --repository-type tfsgit --skip-first-run true > /dev/null
+
+    if [[ "${REPOSITORY}" == "sdlf-team" ]]; then
+      echo "Creating pipeline variables for team repository"
+      az pipelines variable create --pipeline-name ${PREFIX}-${REPOSITORY} \
+          --name sdlf-aztoken --value ${TOKEN} --secret true > /dev/null
+      az pipelines variable create --pipeline-name ${PREFIX}-${REPOSITORY} \
+          --name sdlf-team-firstTime --value "1" > /dev/null
+      az pipelines variable create --pipeline-name ${PREFIX}-${REPOSITORY} \
+          --name sdlf-azure-prefix --value ${PREFIX} > /dev/null
+    fi
+    echo "Running pipelines"
+    for BRANCH in master dev test; do
+      PIPELINE_ID=$(az pipelines run --name ${PREFIX}-${REPOSITORY} --branch ${BRANCH} --folder-path "sdlf" | jq -r .id)
+      if [[ "${REPOSITORY}" == "sdlf-team" ||  "${REPOSITORY}" == "sdlf-foundations" ]]; then # these repositories must be populated synchronously
+        STATUS=$(az pipelines runs show --id ${PIPELINE_ID} | jq -r .status)
+        while [ "$STATUS" != "completed" ]; do
+          echo "Waiting for the pipeline ${PREFIX}-${REPOSITORY} on branch ${BRANCH} to finish" && sleep 7
+          STATUS=$(az pipelines runs show --id ${PIPELINE_ID} | jq -r .status)
+        done
+      fi
+    done
+    if [[ "${REPOSITORY}" == "sdlf-team" ]]; then #disable the firstTime flag to enable teams creation
+      az pipelines variable update --pipeline-name ${PREFIX}-${REPOSITORY} \
+          --name sdlf-team-firstTime --value "0" > /dev/null
+    fi
+} 
+
+function setup_azure_config() {
+    echo "Setting up Azure config and service-connection request"
+    cp -f ${SCM_DIR}/service-connection-template.json ${SCM_DIR}/service-connection.json
+    $SED -i "s|PROJECT|${PROJECT}|g" ${SCM_DIR}/service-connection.json
+    $SED -i "s|AWS_ACCESS_KEY|${AWS_ACCESS_KEY}|g" ${SCM_DIR}/service-connection.json
+    $SED -i "s|AWS_SECRET_KEY|${AWS_SECRET_KEY}|g" ${SCM_DIR}/service-connection.json
+    $SED -i "s|SERVICE_CONNECTION|${SERVICE_CONNECTION}|g" ${SCM_DIR}/service-connection.json
+
+    az devops configure --defaults organization="https://dev.azure.com/$ORG" project="$PROJECT"
+    echo "Creating service-connection \"${SERVICE_CONNECTION}\" on the project: \"${PROJECT}\""
+    SC_ID=$(az devops service-endpoint create --service-endpoint-configuration \
+      ${SCM_DIR}/service-connection.json | jq -r .id)
+    az devops service-endpoint update --id ${SC_ID} --enable-for-all > /dev/null
+}
+
+function deploy_sdlf_foundations_scm() {
+    technical_requirements
+    cd "${DIRNAME}"
+    declare -a REPOSITORIES=("sdlf-team" "sdlf-foundations" "sdlf-pipeline" "sdlf-dataset" "sdlf-datalakeLibrary" "sdlf-pipLibrary" "sdlf-stageA" "sdlf-stageB")
```

Comment:
Would it make sense if we define the `REPOSITORIES` variable once in `deploy.sh` script and then have it be used everywhere?

I am just thinking that we would have to change it in every 3rd party script every time we add/remove repositories otherwise

Reply:
No problem with this. I'll check it

Reply:
Thanks, I think it's just a question of moving this [line](https://github.com/awslabs/aws-serverless-data-lake-framework/blob/3d69adfcb41eca43d24828adcdae90c957ca3256/deploy.sh#L102) out of the function and placing it higher in the `deploy.sh` script